pkg/registry/serving/composer.go

```diff
@@ -79,33 +79,39 @@ func validateComponentSpecs(p *seldonv1.PredictorSpec) error {
 func Compose(sdep *seldonv1.SeldonDeployment) error {
 	sdep.Spec.Name = sdep.ObjectMeta.Name
 
-	for i, p := range sdep.Spec.Predictors {
-		if err := validateComponentSpecs(&p); err != nil {
-			return err
-		}
-
-		if sdep.Spec.Predictors[i].Annotations == nil {
-			sdep.Spec.Predictors[i].Annotations = make(map[string]string)
+	for _, p := range sdep.Spec.Predictors {
+		if !seldonv1.IsPrepack(&p.Graph) {
+			if err := validateComponentSpecs(&p); err != nil {
+				return err
+			}
+
+			// compose user containers
+			composeUserContainer(sdep, &p, &p.ComponentSpecs[0].Spec.Containers[0])
+
+			// compose SeldonPodSpec
+			if err := composeSeldonPodSpec(&p, p.ComponentSpecs[0]); err != nil {
+				return err
+			}
 		}
-		// use no-engine mode
-		sdep.Spec.Predictors[i].Annotations[seldonv1.ANNOTATION_NO_ENGINE] = "true"
-
-		sdep.Spec.Predictors[i].Name = sdep.Spec.Predictors[i].Graph.Name
 
-		// compose user containers
-		composeUserContainer(sdep, &p, &p.ComponentSpecs[0].Spec.Containers[0])
-
-		// compose SeldonPodSpec
-		if err := composeSeldonPodSpec(&p, p.ComponentSpecs[0]); err != nil {
-			return err
-		}
+		setupNoEngineMode(&p)
 	}
 
 	composeInitContainer(sdep)
 
 	return nil
 }
 
+func setupNoEngineMode(pu *seldonv1.PredictorSpec) {
```

Comment:
setupNoEngineMode should set no engine mode, should not set pu.Name, WDYT

Reply:
make sense